File: Sliding Window/0220.Contains Duplicate III/Solution.java
class Solution {
    public boolean containsNearbyAlmostDuplicate(int[] nums, int k, int t) {
        int m = nums.length;
        TreeSet<Integer> ts = new TreeSet<>();
        for (int i = 0; i < m; i++) {
            if (i > k) {
                ts.remove(nums[i - k - 1]);
            }
            Integer bigOne = ts.ceiling(nums[i]);
            if (bigOne != null && Math.abs((long)bigOne - (long)nums[i]) <= t) {
                return true;
            }
            Integer smallOne = ts.floor(nums[i]);
            if (smallOne != null && Math.abs((long)smallOne - (long)nums[i]) <= t) {
                return true;
            }
            ts.add(nums[i]);
        }
        return false;
    }
}

File: Sliding Window/0220.Contains Duplicate III/Solution.py
class Solution:
    def containsNearbyAlmostDuplicate(self, nums: List[int], k: int, t: int) -> bool:
        arr = list()
        for i in range(len(nums)):
            if len(arr) > k:
                arr.remove(nums[i - k - 1])
            arr.append(nums[i])
            list.sort(arr)
            index = bisect.bisect_left(arr, nums[i])
            if index > 0 and abs(arr[index - 1] - arr[index]) <= t:
                return True
            if index < len(arr) - 1 and abs(arr[index + 1] - arr[index]) <= t:
                return True
        return False